src/NUnitCommon/nunit.agent.core/AgentOptions.cs

```diff
@@ -0,0 +1,125 @@
+﻿// Copyright (c) Charlie Poole, Rob Prouse and Contributors. MIT License - see LICENSE.txt
+
+using NUnit.Engine;
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace NUnit.Agents
+{
+    /// <summary>
+    /// All agents, either built-in or pluggable, must be able to 
+    /// handle the options defined in this class. In some cases,
+    /// it may be permissible to ignore them but they should never
+    /// give rise to an error.
+    /// </summary>
+    public class AgentOptions
+    {
+        static readonly char[] DELIMS = new[] { '=', ':' };
+        // Dictionary containing valid options with bool value true if a value is required.
+        static readonly Dictionary<string, bool> VALID_OPTIONS = new Dictionary<string, bool>();
```

Comment:
question: will arg-parsing be case sensitive or case-insensitive? I'm unsure if a `StringComparer.OrdinalIgnoreCase` could be required as a parameter to this Dictionary constructor or not

Reply:
Because NUnit is cross-platform, we have always made arguments case-sensitive.